### PR TITLE
Move to diem

### PR DESCRIPTION
**Pull request by RIg410**

### CODE REVIEW

File: mvm/build.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+fn main() {
+    run("tests/assets", "sh", &["./build_assets.sh"]);
+}
+
+pub fn run(path: &str, cmd: &str, args: &[&str]) {
+    let status = Command::new(cmd)
+        .current_dir(path)
+        .args(args)
+        .stdout(Stdio::piped())
+        .spawn()
```
Comment (vitvakatu) [inline mvm/build.rs:12]: nit: can be replaced with `.status()`

File: mvm/src/io/context.rs [hunk 2]
```diff
@@ -23,13 +21,13 @@ impl ExecutionContext {
     }
 
     pub fn resolve(&self, tag: &StructTag) -> Option<Vec<u8>> {
-        if tag.module.as_str() == TIMESTAMP_MODULE && tag.name.as_str() == CURRENT_TIME_MICROSECONDS {
-             bcs::to_bytes(&self.timestamp).ok()
+        if tag.module.as_str() == TIMESTAMP_MODULE && tag.name.as_str() == CURRENT_TIME_MICROSECONDS
+        {
```
Comment (vitvakatu) [inline mvm/src/io/context.rs:26]: nit: Is it rustfmt style? First time I see this

File: mvm/src/lib.rs [hunk 3]
```diff
@@ -3,16 +3,16 @@
 #[macro_use]
 extern crate alloc;
 extern crate sp_core;
-use crate::data::ExecutionContext;
+use crate::io::context::ExecutionContext;
 use crate::types::{Gas, ModuleTx, PublishPackageTx, ScriptTx, VmResult};
 
-pub mod data;
+//pub mod data;
```
Comment (vitvakatu) [inline mvm/src/lib.rs:9]: nit: do we need it commented?

File: types/src/resources/currency_info.rs [hunk 4]
```diff
@@ -0,0 +1,141 @@
+use crate::{
```
Comment (vitvakatu) [inline types/src/resources/currency_info.rs:1]: Shouldn't we add a copyright notice at the top of the file as in `src/types/event.rs`?

**Review by vitvakatu — COMMENTED** [review]
LGTM overall, a few nitpicks

File: mvm/src/io/context.rs (hunk 2 above)
Comment (RIg410) [reply mvm/src/io/context.rs:26]: Yap. fmt inspection passed.

File: types/src/resources/currency_info.rs (hunk 4 above)
Comment (RIg410) [reply types/src/resources/currency_info.rs:4]: Done

File: mvm/build.rs (hunk 1 above)
Comment (RIg410) [reply mvm/build.rs]: Done.

File: mvm/src/lib.rs (hunk 3 above)
Comment (RIg410) [reply mvm/src/lib.rs]: fixed it.

**Review by borispovod — APPROVED** [review]

**Review by vitvakatu — APPROVED** [review]

**Pull request merged**